source/common/http/filter/ratelimit.cc

```diff
@@ -26,6 +26,19 @@ void ServiceToServiceAction::populateDescriptors(const Router::RouteEntry& route
       {{{"to_cluster", route.clusterName()}, {"from_cluster", config.localServiceCluster()}}});
 }
 
+void RequestHeadersAction::populateDescriptors(const Router::RouteEntry& route,
+                                               std::vector<::RateLimit::Descriptor>& descriptors,
+                                               FilterConfig&, const HeaderMap& headers) {
+  if (headers.has(header_name_)) {
+    if (!route.rateLimitPolicy().rateLimitKey().empty()) {
```

Comment:
Ahh my bad. I understood it as if/else not in addition if the rate_limit_key exists. Code has been updated.